lib/parsers/acorn-extensions.js

```diff
@@ -0,0 +1,191 @@
+"use strict";
+
+const tt = require("acorn").tokTypes;
+const hasOwn = Object.prototype.hasOwnProperty;
+
+exports.fix = function (parser) {
+  // It's not Reify's job to enforce strictness.
+  parser.strict = false;
+
+  // Tolerate recoverable parse errors.
+  parser.raiseRecoverable = noopRaiseRecoverable;
+
+  // Lookahead method.
+  parser.lookAhead = lookAhead;
+
+  // Export-related modifications.
+  parser.parseExport = parseExport;
+  parser.isExportDefaultSpecifier = isExportDefaultSpecifier;
+  parser.parseExportSpecifiersMaybe = parseExportSpecifiersMaybe;
+  parser.parseExportFromWithCheck = parseExportFromWithCheck;
+  parser.parseExportFrom = parseExportFrom;
+  parser.checkExport = checkExport;
+  parser.shouldParseExportDeclaration = shouldParseExportDeclaration;
+};
+
+function noopRaiseRecoverable() {}
+
+function parseExport(node, exports) {
+  this.next();
+  if (this.type === tt.star) {
+    const specifier = this.startNode();
+    this.next();
+    if (this.eatContextual("as")) {
+      // export * as ns from '...'
+      specifier.exported = this.parseIdent(true);
+      node.specifiers = [
+        this.finishNode(specifier, "ExportNamespaceSpecifier")
+      ];
+      this.parseExportSpecifiersMaybe(node);
+      this.parseExportFromWithCheck(node, exports, true);
+    } else {
+      // export * from '...'
+      this.parseExportFromWithCheck(node, exports, true);
+      return this.finishNode(node, "ExportAllDeclaration");
+    }
+  } else if (this.isExportDefaultSpecifier()) {
+    // export def from '...'
+    const specifier = this.startNode();
+    specifier.exported = this.parseIdent(true);
+    node.specifiers = [
+      this.finishNode(specifier, "ExportDefaultSpecifier")
+    ];
+    if (this.type === tt.comma &&
+        this.lookAhead(1).type === tt.star) {
+      // export def, * as ns from '...'
+      this.expect(tt.comma);
+      const specifier = this.startNode();
+      this.expect(tt.star);
+      this.expectContextual("as");
+      specifier.exported = this.parseIdent(true);
+      node.specifiers.push(
+        this.finishNode(specifier, "ExportNamespaceSpecifier")
+      );
```

Comment:
Just curious does the execution of the `this.expect` stuff just above matter? I noticed one was before the `const specifier =....`.